File: problems/src/test/java/LongestSubstringWithoutRepeatingCharactersTest.java
import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LongestSubstringWithoutRepeatingCharactersTest extends SampledTest {
    @Override
    public String getFileName() {
        return "longest_substring_without_repeating_characters.json";
    }

    @Test
    public void test() {
        for (Sample sample : getSamples(Sample[].class)) {
            for (LongestSubstringWithoutRepeatingCharacters solution
                : LongestSubstringWithoutRepeatingCharacters.values()) {
                assertWithMessage(sample.explanation)
                    .that(solution.lengthOfLongestSubstring(sample.input))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class Sample {
        public String input;
        public int output;
        public String explanation;
    }
}

File: typescript/__tests__/problems1_100/longest-substring-without-repeating-characters.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/longest-substring-without-repeating-characters';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    s: string,
  },
  output: number,
  message: string,
}

describe(
    'LongestSubstringWithoutRepeatingCharacters',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        strictEqual(
                            solution.lengthOfLongestSubstring(sample.input.s),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
